File: app/src/main/java/science/apolline/utils/MapStateManager.java
package science.apolline.utils;


import android.content.Context;
import android.content.SharedPreferences;
import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.model.CameraPosition;
import com.google.android.gms.maps.model.LatLng;

/**
 * Created by sparow on 2/26/2018.
 */
public class MapStateManager {
    private static final String LONGITUDE = "longitude";
    private static final String LATITUDE = "latitude";
    private static final String ZOOM = "zoom";
    private static final String BEARING = "bearing";
    private static final String TILT = "tilt";
    private static final String MAPTYPE = "MAPTYPE";
    private static final String PREFS_NAME ="mapCameraState";

    private SharedPreferences mapStatePrefs;

    private String mMapName;

    public MapStateManager(Context context, String mMapName) {
        mapStatePrefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
        this.mMapName = mMapName;
    }

    public void saveMapState(GoogleMap mapMie) {
        SharedPreferences.Editor editor = mapStatePrefs.edit();
        CameraPosition position = mapMie.getCameraPosition();

        editor.putFloat(LATITUDE+"_"+mMapName, (float) position.target.latitude);
        editor.putFloat(LONGITUDE+"_"+mMapName, (float) position.target.longitude);
        editor.putFloat(ZOOM+"_"+mMapName, position.zoom);
        editor.putFloat(TILT+"_"+mMapName, position.tilt);
        editor.putFloat(BEARING+"_"+mMapName, position.bearing);
        editor.putInt(MAPTYPE+"_"+mMapName, mapMie.getMapType());
        editor.apply();
    }

    public CameraPosition getSavedCameraPosition() {
        double latitude = mapStatePrefs.getFloat(LATITUDE+"_"+mMapName, 0);
        if (latitude == 0)
            return null;
        double longitude = mapStatePrefs.getFloat(LONGITUDE+"_"+mMapName, 0);
        LatLng target = new LatLng(latitude, longitude);

        float zoom = mapStatePrefs.getFloat(ZOOM+"_"+mMapName, 0);
        float bearing = mapStatePrefs.getFloat(BEARING+"_"+mMapName, 0);
        float tilt = mapStatePrefs.getFloat(TILT+"_"+mMapName, 0);

        return new CameraPosition(target, zoom, tilt, bearing);
    }

    public int getSavedMapType() {
        return mapStatePrefs.getInt(MAPTYPE, GoogleMap.MAP_TYPE_NORMAL);
    }
}

File: app/src/main/java/science/apolline/utils/MapStateManager.kt
package science.apolline.utils


import android.content.Context
import android.content.SharedPreferences

import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.model.CameraPosition
import com.google.android.gms.maps.model.LatLng

/**
 * Created by sparow on 2/26/2018.
 */
class MapStateManager(context: Context, private val mMapName: String) {

    private val mapStatePrefs: SharedPreferences

    val savedCameraPosition: CameraPosition?
        get() {
            val latitude = mapStatePrefs.getFloat(LATITUDE + "_" + mMapName, 0f).toDouble()
            if (latitude == 0.0)
                return null
            val longitude = mapStatePrefs.getFloat(LONGITUDE + "_" + mMapName, 0f).toDouble()
            val target = LatLng(latitude, longitude)

            val zoom = mapStatePrefs.getFloat(ZOOM + "_" + mMapName, 0f)
            val bearing = mapStatePrefs.getFloat(BEARING + "_" + mMapName, 0f)
            val tilt = mapStatePrefs.getFloat(TILT + "_" + mMapName, 0f)

            return CameraPosition(target, zoom, tilt, bearing)
        }

    val savedMapType: Int
        get() = mapStatePrefs.getInt(MAPTYPE, GoogleMap.MAP_TYPE_NORMAL)

    init {
        mapStatePrefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
    }

    fun saveMapState(mapMie: GoogleMap?) {
        if (mapMie != null) {
            val editor = mapStatePrefs.edit()
            val position = mapMie.cameraPosition

            with(editor) {
                putFloat(LATITUDE + "_" + mMapName, position.target.latitude.toFloat())
                putFloat(LONGITUDE + "_" + mMapName, position.target.longitude.toFloat())
                putFloat(ZOOM + "_" + mMapName, position.zoom)
                putFloat(TILT + "_" + mMapName, position.tilt)
                putFloat(BEARING + "_" + mMapName, position.bearing)
                putInt(MAPTYPE + "_" + mMapName, mapMie.mapType)
                apply()
            }
        }
    }

    companion object {
        private const val LONGITUDE = "longitude"
        private const val LATITUDE = "latitude"
        private const val ZOOM = "zoom"
        private const val BEARING = "bearing"
        private const val TILT = "tilt"
        private const val MAPTYPE = "MAPTYPE"
        private const val PREFS_NAME = "mapCameraState"
    }
}